pkg/clusterapi/aws/actuator.go

```diff
@@ -413,8 +465,118 @@ func (a *Actuator) Exists(machine *clusterv1.Machine) (bool, error) {
 	if err != nil {
 		return false, fmt.Errorf("error getting EC2 client: %v", err)
 	}
+
+	instances, err := a.getRunningInstances(machine, client)
+	if err != nil {
+		return false, err
+	}
+	if len(instances) == 0 {
+		mLog.Debug("instance does not exist")
+		return false, nil
+	}
+
+	// If more than one result was returned, it will be handled in Update.
+	mLog.Debug("instance exists")
+	return true, nil
+}
+
+// updateStatus calculates the new machine status, checks if anything has changed, and updates if so.
+func (a *Actuator) updateStatus(machine *clusterv1.Machine, instance *ec2.Instance, mLog log.FieldLogger) error {
+
+	mLog.Debug("updating status")
+
+	awsStatus := &cov1.AWSMachineProviderStatus{}
+	var err error
+	if machine.Status.ProviderStatus != nil {
+		awsStatus, err = controller.AWSMachineProviderStatusFromClusterAPIMachine(machine)
```

Comment:
May as well have `AWSMachineProviderStatusFromClusterAPIMachine` return a defalut `AWSMachineProviderStatus` when `machine.Status.ProviderStatus` is nil.